### PR TITLE
Refactor to use mustache templates

### PR DESCRIPTION
**Pull request by ekohl**
This makes it easier to customize the generated spec file. It also includes other fixes from https://github.com/dLobatog/npm2rpm/pull/34.

### CODE REVIEW

File: template.mustache [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+%description
+%{summary}
+
+%prep
+{{#BUNDLED}}
+mkdir -p %{npm_cache_dir}
+for tgz in %{sources}; do
+  echo $tgz | grep -q registry.npmjs.org || npm cache add --cache %{npm_cache_dir} $tgz
+done
+%setup -T -q -a {{SOURCES.length-1}} -D -n %{npm_cache_dir}
```
Comment (ekohl) [inline template.mustache]: note to self: this doesn't work

**Review by dLobatog — COMMENTED** [review]
@ekohl https://github.com/dLobatog/npm2rpm/pull/34/ changes the default template, if this is changing again should I review this one directly instead..?

**Comment by ekohl** [comment]
It includes what's in #34 but I think that merging #34 first is better. I can easily rebase on it and fix the remaining issues in this PR.

File: template.mustache [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+%description
+%{summary}
+
+%prep
+{{#BUNDLED}}
+mkdir -p %{npm_cache_dir}
+for tgz in %{sources}; do
+  echo $tgz | grep -q registry.npmjs.org || npm cache add --cache %{npm_cache_dir} $tgz
+done
+%setup -T -q -a {{PROVIDES.length}} -D -n %{npm_cache_dir}
```
Comment (ekohl) [inline template.mustache:43]: I'm not 100% sure about the PROVIDES.length here or if there'll be an off-by-one error.

**Review by ekohl — COMMENTED** [review]
I think this should be mostly OK to go. Anyone mind having a look at the provides comment?

**Comment by ekohl** [comment]
Superseded by https://github.com/dLobatog/npm2rpm/pull/36

**Pull request closed without merging**